kmipclient/client.go

```diff
@@ -758,3 +831,167 @@ func (ex AttributeExecutor[Req, Resp, Wrap]) WithUsageLimit(total int64, unit km
+	}
+	req, err := f(ex.client).Build()
+	if err != nil {
+		ex.err = err
+		return ex
+	}
+	ex.batch = append(ex.batch, req)
+	return ex
+}
+
+// Exec sends the batch to the remote KMIP server, and returns the parsed responses.
+//
+// Parameters:
+//   - opts: Optional BatchOption functions to modify the request message.
+//
+// Returns:
+//   - BatchResult: The results of the batch operations.
+//   - error: An error if the batch request fails.
+func (ex BatchExec) Exec(opts ...BatchOption) (BatchResult, error) {
+	return ex.ExecContext(context.Background(), opts...)
+}
+
+// ExecContext sends the batch to the remote KMIP server, and returns the parsed responses.
+//
+// Parameters:
+//   - ctx: Context for request cancellation and timeout.
+//   - opts: Optional BatchOption functions to modify the request message.
+//
+// Returns:
+//   - BatchResult: The results of the batch operations.
+//   - error: An error if the batch request fails.
+func (ex BatchExec) ExecContext(ctx context.Context, opts ...BatchOption) (BatchResult, error) {
+	if ex.err != nil {
+		return nil, fmt.Errorf("Request initialization failed: %w", ex.err)
+	}
+	resp, err := ex.client.BatchOpt(ctx, ex.batch, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// MustExec is like Exec except it panics if the request fails.
+//
+// Parameters:
+//   - opts: Optional BatchOption functions to modify the request message.
+//
+// Returns:
+//   - BatchResult: The results of the batch operations.
+//
+// Panics:
+//   - If the request fails, this function panics with the error.
+func (ex BatchExec) MustExec(opts ...BatchOption) BatchResult {
+	return ex.MustExecContext(context.Background(), opts...)
+}
+
+// MustExecContext is like Exec except it panics if the request fails.
+//
+// Parameters:
+//   - ctx: Context for request cancellation and timeout.
+//   - opts: Optional BatchOption functions to modify the request message.
+//
+// Returns:
+//   - BatchResult: The results of the batch operations.
+//
+// Panics:
+//   - If the request fails, this function panics with the error.
+func (ex BatchExec) MustExecContext(ctx context.Context, opts ...BatchOption) BatchResult {
+	resp, err := ex.ExecContext(ctx, opts...)
+	if err != nil {
+		//TODO: Add operation ID string
+		panic(fmt.Errorf("Request failed: %w", err))
+	}
+	return resp
+}
+
+// BatchResult represents a collection of KMIP response batch items returned from a KMIP operation.
+type BatchResult []kmip.ResponseBatchItem
+
+// Unwrap checks for eventual errors in all the batch items, and returns an array
+// of item's payloads, and the encountered errors. If an item has no payload, the returned
+// array will contain a nil element at the item index.
+//
+// Returns:
+//   - []kmip.OperationPayload: The slice of operation payloads from the batch result.
+//   - error: An error if any batch item contains an error; otherwise, nil.
+func (br BatchResult) Unwrap() ([]kmip.OperationPayload, error) {
```

Comment:
Why not return a ([]kmip.OperationPayload, error) straight from Exec ?
And regarding MustExec, does it make sense to get a (resp, error) pair if the execution panics on error ?

Reply:
ALternatively I could implement an additional `Exec` variant that does this unwrap, but I couldn't find a good name (I'm also unsatisfied with `Unwrap`)

Reply:
Ok makes sense, I missed the two different level or errors